operator/src/network/ceramic.rs

```diff
@@ -116,15 +119,117 @@ pub fn service_spec() -> ServiceSpec {
 pub struct CeramicSpec {
     /// Name of a config map with a ceramic-init.sh script that runs as an initialization step.
     pub init_config_map: Option<String>,
+    /// Configuration of the IPFS container
+    pub ipfs: Option<IpfsSpec>,
+}
+
+/// Describes how the IPFS node for a peer should behave.
+#[derive(Default, Serialize, Deserialize, Debug, PartialEq, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct IpfsSpec {
+    // Both Go and Rust have the same specification.
+    // If this changes, its possible to convert IpfsSpec into an enum
+    // with variants for each kind.
+    // Until that need arises we use a `kind` field on the struct.
+    pub kind: IpfsKind,
+    pub image: Option<String>,
+    pub image_pull_policy: Option<String>,
+}
+
+#[derive(Default, Serialize, Deserialize, Debug, PartialEq, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub enum IpfsKind {
+    #[default]
+    Rust,
+    Go,
 }
 
 pub struct CeramicConfig {
     pub init_config_map: String,
+    pub ipfs: IpfsConfig,
+}
+
+pub enum IpfsConfig {
+    Rust(RustIpfsConfig),
+    Go(GoIpfsConfig),
+}
+
+impl Default for IpfsConfig {
+    fn default() -> Self {
+        Self::Rust(Default::default())
+    }
 }
+impl IpfsConfig {
+    fn config_maps(&self) -> BTreeMap<String, BTreeMap<String, String>> {
+        match self {
+            IpfsConfig::Rust(_) => BTreeMap::new(),
+            IpfsConfig::Go(config) => config.config_maps(),
+        }
+    }
+    fn into_container(self) -> Container {
+        match self {
+            IpfsConfig::Rust(config) => config.into_container(),
+            IpfsConfig::Go(config) => config.into_container(),
+        }
+    }
+    fn volumes(&self) -> Vec<Volume> {
+        match self {
+            IpfsConfig::Rust(_) => Vec::new(),
+            IpfsConfig::Go(config) => config.volumes(),
+        }
+    }
+}
+
+pub struct RustIpfsConfig {
+    image: String,
+    image_pull_policy: String,
+}
+
+impl Default for RustIpfsConfig {
+    fn default() -> Self {
+        Self {
+            image: "public.ecr.aws/r5b3e0r5/3box/ceramic-one:latest".to_owned(),
+            image_pull_policy: "Always".to_owned(),
+        }
+    }
+}
+impl From<IpfsSpec> for RustIpfsConfig {
+    fn from(value: IpfsSpec) -> Self {
+        let default = RustIpfsConfig::default();
+        Self {
+            image: value.image.unwrap_or(default.image),
+            image_pull_policy: value.image_pull_policy.unwrap_or(default.image_pull_policy),
+        }
+    }
+}
+
+pub struct GoIpfsConfig {
+    image: String,
+    image_pull_policy: String,
+}
+impl Default for GoIpfsConfig {
+    fn default() -> Self {
+        Self {
+            image: "ipfs/kubo:v0.19.1@sha256:c4527752a2130f55090be89ade8dde8f8a5328ec72570676b90f66e2cabf827d". to_owned(),
```

Comment:
I guess this default we would have have to just update manually? would be nice to have defaults get latest used in ceramic

also is there any config output paired with metrics/results? esp for defaults would be nice to know images used for results